### ISSUE TITLE
Output memory config being ignored in batch norm's NHW means

### ISSUE THREAD

**Issue by rmillerTT** [tenstorrent/tt-metal#25198]
### Describe the bug

The output memory config in mean_NHW() (batch_norm.cpp) is initialized, but not passed to ttnn::mean():

```c++
inline Tensor mean_NHW(const Tensor& input_tensor, const std::optional<MemoryConfig>& memory_config) {
    auto output_mem_config = memory_config.value_or(input_tensor.memory_config());
    ttnn::SmallVector<int> dims = {2, 3};
    Tensor mean_hw = ttnn::mean(input_tensor, dims, true);
    return ttnn::mean(mean_hw, 0, true);
}
``` 

Because of this, ttnn::mean() will use whatever tt::tt_metal::operation::DEFAULT_OUTPUT_MEMORY_CONFIG is set to instead of what is provided to the function.

### Steps to reproduce the issue

Can adapt a unit test in test_batch_norm.py to use different memory configs (in training mode) and see that the mean_NWH() operation is using the wrong output memory config (should specify that this has NOT been done, but is suspected)

### Expected behavior

It should use the memory config passed in (or the value_or value)

**Comment by iwroszTT** [comment]
Experiments with a modified batch_norm unit test and some log statements around mean_NHW() confirm that the output memory config is being ignored by the function. As stated in the description of this issue (and observed in the experiments), this may lead to a situation, where the mean_NHW's output would have a different memory config to the one provided by the caller.

This issue seems to be functionally hidden because of the subsequent call to ttnn::square(), which properly handles its memory config param. In consequence, the input tensor passed to the second call of mean_NHW() has the memory config as expected for the output of the operation:
`batch_mean = mean_NHW(input, memory_config, compute_kernel_config);`
`auto mean_sq = mean_NHW(ttnn::square(input, memory_config), memory_config, compute_kernel_config);`

I suspect that we cannot detect this behavior simply by checking the contents or parameters of the final output from the operation, without accessing some lower level APIs, to track memory accesses, etc. For this reason, the fix PR just uses the output_memory_config in the calls made by mean_NHW.
